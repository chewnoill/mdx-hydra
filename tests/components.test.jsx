/** @jest-environment node */
import React from "react";
import styled from "@emotion/styled";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
Enzyme.configure({ adapter: new Adapter() });

const { renderMdx } = require("../build/render-mdx");
const { hydrate } = require("../build/hydrate");
const { renderToString } = require("../build/render-to-string");

const ExampleMDX = `
# This is a test

of the MDX system

<div>This is also a test</div>
`;

const components = {
  h1: styled.h3(`font-size:22px;color: tomato;`),
};

describe("base functionality", () => {
  it("renders MDX correctly", () => {
    const tree = Enzyme.render(
      renderMdx({ source: ExampleMDX, components }).component
    );
    expect(tree).toMatchSnapshot();
  });

  const renderedString = renderToString({ source: ExampleMDX, components });
  it("creates static markup correctly", () => {
    expect(renderedString.staticMDX).toMatchSnapshot();
  });
  const Hydrate = hydrate;
  it("hydrates components correctly", () => {
    const tree = Enzyme.render(
      <Hydrate {...renderedString} components={components} />
    );
    expect(tree).toMatchSnapshot();
  });
});
