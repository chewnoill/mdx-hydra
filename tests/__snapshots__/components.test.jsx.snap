// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base functionality creates static markup correctly 1`] = `"<style data-emotion-css=\\"1gr5sma\\">.css-1gr5sma{font-size:22px;color:tomato;}</style><h3 class=\\"css-1gr5sma\\">This is a test</h3><p>of the MDX system</p><div>This is also a test</div>"`;

exports[`base functionality hydrates components correctly 1`] = `
<div>
  <h3
    class="css-1gr5sma"
  >
    This is a test
  </h3>
  <p>
    of the MDX system
  </p>
  <div>
    This is also a test
  </div>
</div>
`;

exports[`base functionality renders MDX correctly 1`] = `
Array [
  null,
  <h3
    class="css-1gr5sma"
  >
    This is a test
  </h3>,
  <p>
    of the MDX system
  </p>,
  <div>
    This is also a test
  </div>,
]
`;
